api/src/main/java/io/strimzi/api/kafka/model/listener/KafkaListenerAuthenticationCustom.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.api.kafka.model.listener;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import io.strimzi.api.kafka.model.CertSecretSource;
+import io.strimzi.api.kafka.model.Constants;
+import io.strimzi.api.kafka.model.GenericSecretSource;
+import io.strimzi.crdgenerator.annotations.Description;
+import io.sundr.builder.annotations.Buildable;
+import lombok.EqualsAndHashCode;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Configures a listener to use mutual TLS authentication.
+ */
+@Buildable(
+        editableEnabled = false,
+        builderPackage = Constants.FABRIC8_KUBERNETES_API
+)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@EqualsAndHashCode
+public class KafkaListenerAuthenticationCustom extends KafkaListenerAuthentication {
```

Comment:
I decided to leave out `tlsTrustedCertificates` despite it being listed in the proposal, due to the primary reason being the user can easily? define the truststore themself, reference it as a secret, and specify it as a config.

Reply:
Yeah ... that should be fine => I thought that that was also what we wanted in the proposal. Maybe we just forgot to update it. Feel free to remove it from the code as far as I'm concerned.